avocado/utils/cpu.py

```diff
@@ -90,7 +90,7 @@ def cpu_has_flags(flags):
         flags = [flags]
 
     for flag in flags:
-        if not _list_matches(cpu_info, '.*%s.*' % flag):
+        if not any([flag.encode() in line for line in cpu_info]):
```

Comment:
cpu_info() is returning not only flags but all info by CPUs. This could lead to false positives, right? Would it be better to check only on the line starting with `flags\t\t:`? For instance, the flag `ss` is present on the line `address sizes`.

Reply:
It could, and it's a *different* issue that I'm already preparing a patch :)

Reply:
Basically, I think we should be even more rigorous, and besides spaces, we should check that info on specific lines of `/proc/cpuinfo`, but that depends on the arch. For i386 and x86_64, it's `flags: `, for s390x, it should be `features: `, etc.

Reply:
I see. But we could get only the lines that start with `flags:` and `features:`. Yea, anyway.. I just t think that searching for the entire cpu_info is "dangerous".

Reply:
I agree, and it's surprising that this is the current implementation. We need to step up the game here, but I think it's better suitable post-release, and we can discuss the new terms of a new API and implementation.

Reply:
> I see. But we could get only the lines that start with `flags:` and `features:`. Yea, anyway.. I just t think that searching for the entire cpu_info is "dangerous".

We have to check the other arches too... I just listed two as examples. Power 8 shows:

```
cpu             : POWER8E (raw), altivec supported
```

So we have to decide if `cpu_has_flags('altivec')` would work on a new implementation.

Reply:
```
      cpu_flags = [l for l in cpu_info if l.startswith(('flags'.encode(), 'features'.encode()))]
```

Reply:
Hummm I see. yes, lets pospone.